### PR TITLE
Cleanup of 'Merge-Hashtable'

### PR DESCRIPTION
**Pull request by raandree**
Also assigned type [hashtable] to 'ReferenceHashtable' and 'DifferenceHashtable' and made the parameters mandatory. @gaelcolas, do you think assigning the type breaks something that we haven't covered by the tests?

### CODE REVIEW

**Comment by gaelcolas** [comment]
Yeah, it was commented out `# [hashtable] These should stay ordered` because if you cast the `[OrderedDictionary]` back to `[hashtable]` then you do not guarantee the order for next manipulation or comparison...
I'm not saying it'll break, but it's very likely that it does in some weird ways.
At some point we'll probably want to use [IDictionary] or something, but that wouldn't fly with PSCustomObject.
I'd say just leave as [Object] for now?

File: source/Private/Merge-Hashtable.ps1 [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 {
-    [outputType([hashtable])]
-    [cmdletBinding()]
-    Param(
+    [OutputType([hashtable])]
+    [CmdletBinding()]
+    param (
         # [hashtable] These should stay ordered
+        [Parameter(Mandatory = $true)]
+        [hashtable]
```
Comment (gaelcolas) [inline source/Private/Merge-Hashtable.ps1:8]: too risky

File: source/Private/Merge-Hashtable.ps1 [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
         $ReferenceHashtable,
 
         # [hashtable] These should stay ordered
+        [Parameter(Mandatory = $true)]
+        [hashtable]
```
Comment (gaelcolas) [inline source/Private/Merge-Hashtable.ps1:13]: too risky, use [object] instead?

**Review by gaelcolas — CHANGES_REQUESTED** [review]

File: source/Private/Merge-Hashtable.ps1 (hunk 2 above)
Comment (raandree) [reply source/Private/Merge-Hashtable.ps1:13]: Done

**Review by gaelcolas — APPROVED** [review]

**Pull request merged**